python-avd/tests/pyavd/j2filters/test_convert_dict.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import absolute_import, division, print_function
+
+import pytest
+from pyavd.j2filters.convert_dicts import convert_dicts
+
+DEFAULT_PRIMARY_KEY = "name"
+NESTED_LIST_OF_DICT = {
+    "TEST1": [{"type": "permit", "extcommunities": "65000:65000"}, {"type": "deny", "extcommunities": "65002:65002"}],
+    "TEST2": [{"type": "deny", "extcommunities": "65001:65001"}],
```

Comment:
can we have one more level of nested dict, is this supported?

Reply:
"TEST1": [{"type": "permit", "extcommunities": "65000:65000", "more": {"data": true}}, {"type": "deny", "extcommunities": "65002:65002"}]

Reply:
This function will insert primary_key to outer dictionary keys.